Add Karma tests, travis config.

dmitrizagidulin opened the pull request:
Refactor to run as karma tests.

dmitrizagidulin commented:
BREAKING: The import signature on this lib is changed in this PR.

Before:
```
const X25519KeyPair = require('x25519-key-pair');
```

Now:
```
const {X25519KeyPair} = require('x25519-key-pair');
```

dlongley commented:
This is probably good but we can't merge it yet without disrupting some things. I'll review more closely and approve when we can merge.

dmitrizagidulin commented:
@dlongley I agree (re disruption). Though I think the way to fix that is:

1. Merge `initial` into `master`, and publish `v0.1.0`.
2. Rebase THIS PR onto `master` (and optionally publish it as `v0.2.0`).

That way, the `initial` branch (on which stuff is depending) will be left alone.

dlongley commented:
@dmitrizagidulin -- that would be fine, but there's a high likelihood that someone will delete the initial branch when you do that, even if we put protection on it; because the pattern is... once released, delete `initial`. We can probably merge this at the beginning of next week.

dlongley commented:
@davidlehn -- I'd like us to be able to pull this in today. Could you make sure anything using the old branches will get updated?

davidlehn commented:
@dlongley Ok, is there a list of dependents?

davidlehn commented on tests/unit/index.js:
```diff
@@ -65,7 +65,7 @@ describe('X25519KeyPair', () => {
       const result = key.verifyFingerprint(fingerprint);
       expect(result.valid).to.be.true;
-      expect(result.error).to.not.exist;
+      // expect(result.error).to.not.exist;
```
Why is this commented out?  Seems it should be enabled or removed.

dmitrizagidulin replied:
Fixed.

davidlehn commented on tests/package.json:
```diff
@@ -0,0 +1,36 @@
+  "author": {
+    "name": "Digital Bazaar, Inc.",
+    "email": "support@digitalbazaar.com",
+    "url": "http://digitalbazaar.com/"
```
```suggestion
    "url": "https://digitalbazaar.com/"
```

dmitrizagidulin replied:
Fixed.

davidlehn commented on .travis.yml:
```diff
@@ -0,0 +1,18 @@
+language: node_js
+node_js:
+  - "8"
+  - "10"
```
Add 12

dmitrizagidulin replied:
Added

dmitrizagidulin commented:
Ok, now just gotta fix the `TypeError: publicKeyBuffer.equals is not a function` error in the browser.

davidlehn commented:
That's the line that had the conflict?  Perhaps that's why the other version was there?

dmitrizagidulin commented:
@davidlehn No, the conflict line was right before that.

This error is our old friend - Buffers in node.js has the .equals() method, but whatever the return type of base58.decode() from forge is, doesn't have an .equals() method.

davidlehn commented:
@dmitrizagidulin I'm suggesting the other version was there to avoid the buffer issue:
```diff
-    const valid = fingerprintBuffer.slice(0, 1).toString('hex') === '25' &&                
-      publicKeyBuffer.equals(fingerprintBuffer.slice(1));                                  
+    const valid = fingerprintBuffer[0] === 37 &&                                           
+      publicKeyBuffer.toString() === fingerprintBuffer.slice(1).toString();                

```

dmitrizagidulin commented:
@davidlehn Doh, I see... I don't remember why I changed it in the first place! (I _thought_ it was breaking some test?)

Anyways, reverted, both Node and Karma tests now work. We should be ready to merge.

mattcollier commented:
Tests are passing locally and on CI.  Merging.
